agent/command.go

```diff
@@ -251,6 +251,19 @@ func (a *Agent) runCommand(ctx context.Context, tc *taskContext, commandInfo mod
 		tc.setCurrentIdleTimeout(cmd)
 	}
 	a.comm.UpdateLastMessageTime()
+	defer func() {
+		// After this block is done running, add the command cleanups to the
+		// task context from the task config and then clear the task config's
+		// command cleanups. Setup group commands are handled differently and
+		// should only be cleaned up after the entire group has run while
+		// any other block should clean up after the task is done running.
+		if options.block == command.SetupGroupBlock {
+			tc.addSetupGroupCommandCleanups(tc.taskConfig.CommandCleanups)
+		} else {
+			tc.addTaskCommandCleanups(tc.taskConfig.CommandCleanups)
+		}
+		tc.taskConfig.CommandCleanups = nil
```

Comment:
Since the other `taskConfig` methods use a lock, should we use a lock when unsetting it as well?

I don't think a lock is necessary in either case since commands run serially, but making it one way or the other (either all locked or all unlocked) would ensure it behaves consistently.